### ISSUE TITLE
Works Fine on Ubuntu 19.04 Laptop, Does not on Windows 10 WSL Desktop

### ISSUE THREAD

**Issue by bmw417** [img2tab/macos-guest-virtualbox#61]
The script worked flawlessly on Linux, but since the README also listed WSL support I figured I'd give it a go installing it on my Windows 10 desktop using the WSL (Ubuntu 18). However, the error I get is:

```
Can't find VBoxManage in PATH variable,
checking C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
Found VBoxManage
Please make sure Oracle VM VirtualBox Extension Pack is installed, and that
all installed VirtualBox extensions are listed as usable in "VBoxManage list extpacks"
```

I've installed the extension pack and it is visible when I pull it up in PowerShell:

```
PS C:\Windows\system32> & 'C:\Program Files\Oracle\VirtualBox\VBoxManage.exe' list extpacks
Extension Packs: 1
Pack no. 0:   Oracle VM VirtualBox Extension Pack
Version:      6.0.8
Revision:     130520
Edition:
Description:  USB 2.0 and USB 3.0 Host Controller, Host Webcam, VirtualBox RDP, PXE ROM, Disk Encryption, NVMe.
VRDE Module:  VBoxVRDP
Usable:       true
Why unusable:
```

Any suggestions / something I'm doing wrong?

**Comment by img2tab** [comment]
Thanks for the bug report! Definitely a bug in the script. Will investigate.

**Comment by img2tab** [comment]
@bmw417 I've updated the script and tested on WSL on Windows 10 1903 18362.116. macOS installed successfully.

Please test [macos-guest-virtualbox.sh version 0.62.0](https://raw.githubusercontent.com/img2tab/macos-guest-virtualbox/98a9e9d05e049b09eafe3824c0a3121c53f57472/macos-guest-virtualbox.sh)

**Comment by bmw417** [comment]
It seems like I may be missing something, since I get that same error. However, it makes complete sense, since the error is here: 

'''
# Oracle VM VirtualBox Extension Pack
extpacks="$(VBoxManage list extpacks 2>/dev/null)"
if [ "$(expr match "${extpacks}" '.*Oracle VM VirtualBox Extension Pack')" -le "0" \
    -o "$(expr match "${extpacks}" '.*Usable:[[:blank:]]*false')" -gt "0" ]; then
    echo "Please make sure Oracle VM VirtualBox Extension Pack is installed, and that"
    echo "all installed VirtualBox extensions are listed as usable in \"VBoxManage list extpacks\""
    exit
fi
'''

I have VBox 6.0.6 downloaded on Windows, so you obviously can't run VBoxManage.exe from bash. I just decided to delete the check (knowing that it'll probably fail elsewhere), but then kept it going. It looks like after it finds VBoxManage.exe, it still tries to run the VBoxManage Linux binary, which won't work. Maybe I misunderstand the scope of the script, so what I thought it could do was also create a MacOS VM for Windows as well using the WSL to run the script, but I may have just misunderstood. What's your setup for the WSL? Are you running a VM off of a graphical environment like [this?](https://www.reddit.com/r/Windows10/comments/8dnyig/using_a_gui_with_wsl_on_windows_10_ubuntu_wsl/)

**Comment by img2tab** [comment]
Before that error message you should get the following output:
```
Press enter to continue, CTRL-C to exit.
checking /mnt/c/Program Files/Oracle/VirtualBox/VBoxManage.exe
Found VBoxManage
````
The script should exit if it doesn't find `/mnt/c/Program Files/Oracle/VirtualBox/VBoxManage.exe`, so I don't understand quite where the bug is at the moment.

The v0.62.0 script should work on WSL whether or not it's running in a graphical environment because Windows exposes `/mnt/c/` to WSL either way.

**Comment by bmw417** [comment]
I'm such an idiot. I was a bit tired last night and working between three systems, and happened to do a `sudo apt-get install virtualbox` in WSL instead of my other computer, which tripped your script up since there were two virtualbox binaries running around. After purging it and re-rerunning, it's started the interaction process in Windows. Since I'm on a Ryzen system however, I'll have to figure out which cpu profile to use. 

Great script and great job! I'll mark as resolved, and I apologize again for my mistake.

**Comment by img2tab** [comment]
Oh I see... if you have `VBoxManage` in the WSL path then the following `if` statement silently fails (line 128):
`if [ -z "$(VBoxManage -v 2>/dev/null)" ]; then`
This is actually a bug in the script and I'll fix it in the next version with this:
`if [ -z "$(VBoxManage.exe -v 2>/dev/null)" ]; then`
Thanks for testing!

**Comment by bmw417** [comment]
Absolutely, and great job again!

**Comment by img2tab** [comment]
Fixed in version 0.63.0

**Comment by bmw417** [comment]
One more thing, it seems as though the script is attempting to write the .vdi files to C:\Windows\system32, is there a way to change the path? I'll dig around the script while I wait too
